### ISSUE TITLE
Twitch integration stopped working when the GameEvent system got ripped out

### ISSUE THREAD

**Issue by ShadowBrian** [Facepunch/sbox-issues#4183]
### Describe the bug

Twitch integration stopped working when the GameEvent system got ripped out

Also you have to "exit" your scene game to log in to twitch but now that there's no longer a "go to game menu" button you can't get back into the scene system after

### To Reproduce

1. Try to use twitch integration
2. Realize it doesn't work because of events not existing anymore
3. depression

### Expected behavior

1. Try to use twitch integration
2. It works fine
3. happiness

**Comment by garrynewman** [comment]
Lets think of a better way to implement this that doesn't involve the gameevent system. An interface seems more appropriate maybe?

Or maybe instead of an interface it would be better as a base component?

**Comment by MrBrax** [comment]
So events as a concept are completely being abandoned?

**Comment by garrynewman** [comment]
Mostly yes. They're too messy, too global.

**Comment by MrBrax** [comment]
Wouldn't a pluggable i/o twitch node-thing fit with the scene system? A component then i suppose. I don't entirely understand the new stuff yet. I guess it could become messy with hardcoded output amounts.

**Comment by garrynewman** [comment]
I don't know what any of that means

**Comment by MrBrax** [comment]
Yeah, you guys are way better at figuring this stuff out. I just like node-based workflows.

**Comment by ShadowBrian** [comment]
Yeah for sure, I wasn't suggesting to re-implement it using events, just noting that it stopped working entirely because of them missing currently.
I'm not sure it fits as a component too well since every games use of chat commands and such can vary, but some kind of interface might make sense for sure.
